File: C/beadsort/gemma2-9b-it/gemma2-9b-it.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_VALUE 10000

void bead_sort(int arr[], int n) {
    int max = arr[0];
    for (int i = 1; i < n; i++) {
        if (arr[i] > max) {
            max = arr[i];
        }
    }

    int *beads = (int *)malloc((max + 1) * sizeof(int));
    for (int i = 0; i <= max; i++) {
        beads[i] = 0;
    }

    for (int i = 0; i < n; i++) {
        beads[arr[i]]++;
    }

    int index = 0;
    for (int i = 0; i <= max; i++) {
        for (int j = 0; j < beads[i]; j++) {
            arr[index++] = i;
        }
    }

    free(beads);
}

int main() {
    int arr[1000];
    for (int i = 0; i < 1000; i++) {
        arr[i] = rand() % MAX_VALUE;
    }

    bead_sort(arr, 1000);

    return 0;
}

File: Java/beadsort/gemma2-9b-it/Gemma2_9b_it.java
import java.util.Random;
public class Gemma2_9b_it {

    public static void beadSort(int[] arr) {
        int max = getMax(arr);
        int numBeads = max + 1;

        // Create an array of beads, initially all empty
        boolean[] beads = new boolean[numBeads];

        // Place beads on the array based on the values in the input array
        for (int value : arr) {
            beads[value] = true;
        }

        // Iterate through the beads array and place values back into the input array
        int index = 0;
        for (int i = 0; i < numBeads; i++) {
            if (beads[i]) {
                arr[index++] = i;
            }
        }
    }

    private static int getMax(int[] arr) {
        int max = arr[0];
        for (int value : arr) {
            if (value > max) {
                max = value;
            }
        }
        return max;
    }

    public static void main(String[] args) {
        Random rand = new Random();
        int[] array = new int[1000];

        // Preenchendo o array com valores aleatórios
        for (int i = 0; i < array.length; i++) {
            array[i] = rand.nextInt(1000); // Números entre 0 e 999
        }
        // Ordenando com BeadSort
        Gemma2_9b_it.beadSort(array);
    }
}